starwars_api/models.py

```diff
@@ -28,71 +28,92 @@ def all(cls):
         later in charge of performing requests to SWAPI for each of the
         pages while looping.
         """
-        pass
+        return cls.dispatcher('create_query_set')()
 
 
-class People(BaseModel):
-    """Representing a single person"""
-    RESOURCE_NAME = 'people'
+class OperationsForPeople(object):
+
+    @classmethod
+    def dispatcher(cls, fun):
+        return { 'get_resource_from_api': lambda x: api_client.get_people(x),
+                 'create_query_set':      lambda: PeopleQuerySet(),
+                 'max_resources':         lambda: api_client.get_people()['count'],
+               }[fun]
+
+
+class OperationsForFilms(object):
+
+    @classmethod
+    def dispatcher(cls, fun):
+        return { 'get_resource_from_api': lambda x: api_client.get_films(x),
+                 'create_query_set':      lambda: FilmsQuerySet(),
+                 'max_resources':         lambda: api_client.get_films()['count'],
+               }[fun]
+
 
+class People(BaseModel, OperationsForPeople):
+    """Representing a single person"""
+    
     def __init__(self, json_data):
         super(People, self).__init__(json_data)
 
     def __repr__(self):
         return 'Person: {0}'.format(self.name)
 
 
-class Films(BaseModel):
-    RESOURCE_NAME = 'films'
+class Films(BaseModel, OperationsForFilms):
 
     def __init__(self, json_data):
         super(Films, self).__init__(json_data)
-
+        
     def __repr__(self):
         return 'Film: {0}'.format(self.title)
 
 
 class BaseQuerySet(object):
-
+    
     def __init__(self):
-        pass
-
+        self.counter = 0
+            
     def __iter__(self):
-        pass
+        self.counter = 0
+        return self
 
     def __next__(self):
-        """
-        Must handle requests to next pages in SWAPI when objects in the current
-        page were all consumed.
-        """
-        pass
+        if self.counter < self.dispatcher('max_resources')():
+            self.counter += 1
+            return self.__class__.resource_class.get(self.counter)
+        raise StopIteration()
```

Comment:
So you make a request to the API for the range of the QuerySet length? I was thinking of taking this approach but it seemed more efficient to go through the already downloaded QuerySet and load next page only when the of page is reached.